core/dataset.cpp

```diff
@@ -310,6 +310,44 @@ void Dataset::setData(const std::string &name, Variable data) {
     rebuildDims();
 }
 
+/// Set (insert or replace) data from a DataArray with a given name, avoiding
+/// copies where possible by using std::move.
+///
+void Dataset::setDataMove(const std::string &name, DataArray data) {
+  // Get the Dataset holder
+  auto dataset = DataArray::to_dataset(std::move(data));
+
+  for (const auto &[dim, coord] : dataset.m_coords) {
+    if (const auto it = m_coords.find(dim); it != m_coords.end())
```

Comment:
Any reason for not using `contains`?

Reply:
No reason, apart from the fact that I didn't write this, just copied it from what was already there, but i'll update it

Reply:
as you pointed out, using the iterator avoid having to search twice, to i left it as is